caffe2/contrib/tensorrt/trt_utils.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "caffe2/core/logging.h"
+#include <NvInfer.h>
+#include <iostream>
+
+namespace caffe2 {
+
+  // Logger for GIE info/warning/errors
+class TrtLogger : public nvinfer1::ILogger {
+  using nvinfer1::ILogger::Severity;
+
+ public:
+  TrtLogger(Severity verbosity = Severity::kWARNING) : _verbosity(verbosity) {}
+  void log(Severity severity, const char* msg) override {
+    if (severity <= _verbosity) {
+      if (severity == Severity::kINTERNAL_ERROR || severity == Severity::kERROR) {
+        LOG(ERROR) << msg;
+      } else if (severity == Severity::kWARNING) {
+        LOG(WARNING)  << msg;
+      } else if (severity == Severity::kINFO) {
+        LOG(INFO) << msg;
+      }
+    }
+  }
+
+ private:
+  Severity _verbosity;
+};
+
+struct InferDeleter {
+  template <typename T>
+  void operator()(T* obj) const {
+    if (obj) {
+      obj->destroy();
+    }
+  }
+};
+
+template <typename T>
+inline std::shared_ptr<T> InferObject(T* obj) {
```

Comment:
should it be in another namespace? `InferObject` sounds cryptic

Reply:
OK. I'll put it in a seperate namespace and rename the method.